### PR TITLE
Php8 support

### PR DESCRIPTION
**Pull request by Nielsvanpach**
- [x] Moved from Travis CI to Github Actions
- [x] Added support for PHP 8.0 (which launches today)

In order to support PHP 8, we have to upgrade PHPUnit to v9. This is the only version supporting PHP 8.0. I had to refactor the tests a bit because of incompatibilities and deprecations.
This also means we have to require php ^7.3.

It would be great to see this package support PHP 8!

Also, no green checkmark shown here but you can see the tests passed on my fork: https://github.com/Nielsvanpach/iso3166/commits/php8-support

### CODE REVIEW

**Comment by alcohol** [comment]
Thanks a lot for your time and work; I will try to make some time tonight myself to take a closer look.

**Comment by Propaganistas** [comment]
Looking forward to this.

File: src/Exception/ISO3166Exception.php [hunk 1]
```diff
@@ -1,5 +1,12 @@
 <?php
 
+/*
+ * (c) Rob Bast <rob.bast@gmail.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
```
Comment (pimjansen) [inline src/Exception/ISO3166Exception.php:8]: Dont think it matches the rest of the lib to add this?

File: src/Guards.php [hunk 2]
```diff
@@ -24,9 +24,7 @@ final class Guards
     public static function guardAgainstInvalidName($name)
     {
         if (!is_string($name)) {
-            throw new InvalidArgumentException(
-                sprintf('Expected $name to be of type string, got: %s', gettype($name))
-            );
+            throw new InvalidArgumentException(sprintf('Expected $name to be of type string, got: %s', gettype($name)));
```
Comment (pimjansen) [inline src/Guards.php:27]: Why did we change this? Its a pretty unrelated styling change

File: src/ISO3166DataValidator.php [hunk 3]
```diff
@@ -28,8 +26,6 @@ public function validate(array $data)
     }
 
     /**
-     * @param array $entry
```
Comment (pimjansen) [inline src/ISO3166DataValidator.php:31]: Whats the use case to remove this where the docs are valid?

**Review by pimjansen — COMMENTED** [review]
Overall imo it looks pretty solid but a lot of unrelated style changes

**Comment by Nielsvanpach** [comment]
@pimjansen I did not make any style changes myself. These were done by the php-cs fixer action, which uses the config file .php_cs.dist that was already present in this repo. I assume the command was not run in a while.

**Review by pimjansen — APPROVED** [review]

File: src/ISO3166DataValidator.php (hunk 3 above)
Comment (alcohol) [reply src/ISO3166DataValidator.php:31]: I'm actually ok with removing docs that cover things that are already covered by type hints now a days
